chromadb/api/types.py

```diff
@@ -84,6 +84,9 @@ def validate_ids(ids: IDs) -> IDs:
     for id in ids:
         if not isinstance(id, str):
             raise ValueError(f"Expected ID to be a str, got {id}")
+    if len(ids) != len(set(ids)):
+        dups = set([x for x in ids if ids.count(x) > 1])
```

Comment:
nit: can we do this in one pass over the data

Reply:
We can, my thought was only to do the extra work of building a new set for reporting if there was actually an error so we don't pay for that in the "happy path."